src/Map/index.js

```diff
@@ -121,44 +134,58 @@ export default class Map extends Component {
     }
   }
 
-  loadAddresses = async () => {
-    let { loaded } = this.state
-
-    let {
+  async loadAddresses(prevPops) {
+    const {
       apiKey,
       markerType,
       markerCollection,
       markers: { markerAddress },
-    } = this.props
+      editor,
+    } = prevPops
+    const { loaded } = this.state
 
-    let addr =
+    if (!markerCollection || loaded || editor) {
+      return
+    }
+
+    const locations =
       markerType === 'simple'
         ? markerAddress
           ? [markerAddress]
           : []
         : markerCollection
-        ? markerCollection.map((m) => m.markers_list.markerAddress)
+        ? markerCollection.map(m => m.markers_list.markerAddress)
         : []
-
-    if (!loaded) {
-      let result = await axios.post(geocodeURL, {
-        addresses: addr,
-        key: apiKey,
-      })
-
-      this.setState({
-        addresses: result.data,
-      })
-
-      if (
-        markerType === 'simple' ||
-        (markerType !== 'simple' && markerCollection)
-      ) {
-        this.setState({
-          loaded: true,
+    
+    const coordinates = []
+    const addresses = []
+
+    for (const location of locations) {
+      if (COORD_REG_EX.test(location)) {
```

Comment:
I have no reason to doubt the correctness of this logic for detecting and parsing coordinates. That said, as a reader, it's also pretty hard for me to be confident that it is correct.

What do you think about pulling this out to a small utility module, maybe `isCoordinates(string)` and `parseCoordinates(string)`, and then hitting it with some unit tests to demonstrate correctness? This might also help when it comes time to extend support to different coordinate formats in future.

Reply:
We do not have a testing framework set up for components. I believe @scott-massey was working on that at some point.

Reply:
What if this logic was pulled out of the component?

Reply:
I did not realize what repo this was. 👍

Reply:
(not blocking) Maybe it's still worth asking: how difficult would it be to add `jest` and write some tests for this, even if it wasn't setup on CI just yet? Would now be as good a time as any to add unit tests to this repo?

Reply:
Seems like a good cool-down project. If I get it done before this is merged I can incorporate it into this card.